compute_tools/src/compute.rs

```diff
@@ -283,7 +322,15 @@ impl TryFrom<ComputeSpec> for ParsedSpec {
             timeline_id,
             endpoint_storage_addr,
             endpoint_storage_token,
-        })
+        };
+
+        // Now check validity of the parsed specification
+        match res.validate() {
+            Ok(_) => (),
+            Err(msg) => return Err(msg.clone()),
+        };
```

Comment:
Since try_from also returns result, most likely you can just do 
```suggestion
        res.validate()?
```
here.

Reply:
Apparently that's not possible. Still discovering specific aspects of the Rust language, will appreciate pointers to docs and/or being lectured ;-)

```
    = note: `?` operator cannot convert from `&compute::ParsedSpec` to `std::result::Result<compute::ParsedSpec, std::string::String>`
    = note:   expected enum `std::result::Result<compute::ParsedSpec, std::string::String>`
            found reference `&compute::ParsedSpec`
```

Reply:
I think you can change `validate` to return nothing on success, i.e.

```
pub fn validate(&self) -> Result<(), String> {
...
}
```

Then, here, you can do:
```
res.validate()?;
Ok(res)
```

Reply:
> . Still discovering specific aspects of the Rust language, will appreciate pointers to docs and/or being lectured ;-)

The rust compiler is usually pretty good at lecturing at you :-D

Reply:
I changed the validate signature per-review, looks better now, thanks! And yes the rust compiler seems to have been design for lecturing its users ;-)